### PR TITLE
prctl compile-time fixes, also chasing lints

### PR DESCRIPTION
**Pull request by yaleman**
Fixes:

- the issues with not handling the prctl `Result`
- my innate need to sate clippy
- should help fix issues with missing wasm-related tools on dev machines

Checklist

- [x] This pr contains no AI generated code
- [x] cargo fmt has been run
- [x] cargo clippy has been run
- [x] cargo test has been run and passes
- [ ] book chapter included (if relevant)
- [ ] design document included (if relevant)

### CODE REVIEW

File: server/lib/src/be/mod.rs [hunk 1]
```diff
@@ -1923,8 +1923,8 @@ impl<'a> BackendWriteTransaction<'a> {
         })?;
         Ok(nsid)
     }
-
     pub fn get_db_s_uuid(&mut self) -> Result<Uuid, OperationError> {
+        #[allow(clippy::blocks_in_conditions)]
```
Comment (Firstyear) [inline server/lib/src/be/mod.rs:1927]: We should make the value / block assign to a value instead and then match.

File: server/lib/src/entry.rs [hunk 2]
```diff
@@ -358,6 +358,8 @@ impl Entry<EntryInit, EntryNew> {
     pub(crate) fn unsafe_from_entry_str(es: &str) -> Self {
         // Just use log directly here, it's testing
         // str -> proto entry
+
```
Comment (Firstyear) [inline server/lib/src/entry.rs:361]: I swear this function is one of my deepest regrets.

**Review by Firstyear — CHANGES_REQUESTED** [review]
You may be able to set:
https://doc.rust-lang.org/clippy/lint_configuration.html#allow-unwrap-in-tests 
in lib.rs to make this easier.

File: server/lib/src/entry.rs (hunk 2 above)
Comment (yaleman) [reply server/lib/src/entry.rs:361]: we'll excise it one day!

File: unix_integration/src/unix_passwd.rs [hunk 3]
```diff
@@ -99,14 +99,18 @@ wheel:x:481:admin,testuser
 
     #[test]
     fn test_parse_passwd() {
-        for record in parse_etc_passwd(EXAMPLE_PASSWD.as_bytes()).unwrap() {
+        #[allow(clippy::expect_used)]
```
Comment (Firstyear) [inline unix_integration/src/unix_passwd.rs:102]: Shouldn't this use the allow-expect-in-tests thing?

**Review by Firstyear — APPROVED** [review]

File: clippy.toml [hunk 4]
```diff
@@ -13,3 +13,7 @@ too-many-arguments-threshold = 8
 type-complexity-threshold = 300
 
 disallowed-types = ["std::collections::HashMap", "std::collections::HashSet"]
+
+allow-expect-in-tests = true
+allow-unwrap-in-tests = true
+allow-dbg-in-tests = true
```
Comment (Firstyear) [inline clippy.toml:19]: I have to remember this macro exists. It's good.

**Review by Firstyear — APPROVED** [review]

**Pull request merged**